compress.cc

```diff
@@ -90,6 +93,46 @@ shared_ptr<compressor> compressor::create(const std::map<sstring, sstring>& opti
     return {};
 }
 
+static constexpr auto uncompressed_size_marker_size = 4u;
+
+/*
+ * This is OCD level stuff, but because we for historical reasons (i.e. cassandra sstable compatibility)
+ * write uncompressed size in some compressors (hello lz4), I cannot in good concience allow a user (me)
+ * to duplicate this info for other. So we keep track per compressor type etc whether and how to add
+ * this to a packet.
+ * 
+ * Note: for the fragmented compressor methods above, for some compressors at least we should not need
+ * having this info at all. Esp. zstd/deflate can just stream compressed data until exhausted. LZ4
+ * interface for this is yukky, but workable etc. So room for better fruits here....
+ * 
+ * TLDR: TODO: refactor sstable/other compression to 
+ *      a.) do always fragmented operation
+ *      b.) only use frame markers for lz4 in sstable and interop reasons.
+ */ 
+static size_t write_uncompressed_size(int8_t* out, size_t size, size_t input_len) {
+    if (size < uncompressed_size_marker_size) {
+        throw std::invalid_argument("Buffer underflow");
+    }
+    out[0] = input_len & 0xFF;
+    out[1] = (input_len >> 8) & 0xFF;
+    out[2] = (input_len >> 16) & 0xFF;
+    out[3] = (input_len >> 24) & 0xFF;
+
+    return uncompressed_size_marker_size;
```

Comment:
read_be()?

Reply:
I mean write_be().

Reply:
Indeed, however, for consistency/continuity sake I copied/moved the code from `lz4_processor::compress`. Since we no longer